From 7ca180e6559e974bc76bd028178e8fd49cd1d694 Mon Sep 17 00:00:00 2001
From: Yehuda Yitschak <yehuday@marvell.com>
Date: Wed, 16 Jul 2014 11:46:09 +0300
Subject: [PATCH 0039/1240] config: Some more cleanups on config files

	- Changed the identity strings
	- Moved ICACHE definition to mvebu-common
	- Moved bootargs to mvebu-common
	- Add override for bootargs in A8K

Change-Id: Ic5c270fe6a7ed846adce427d98cb32be538952c4
Signed-off-by: Yehuda Yitschak <yehuday@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/9341
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
---
 include/configs/armada8k.h     | 30 +++++++++++++++---------------
 include/configs/mvebu-common.h | 12 +++++++++++-
 2 files changed, 26 insertions(+), 16 deletions(-)

diff --git a/include/configs/armada8k.h b/include/configs/armada8k.h
index 4fbdee5..1508025 100644
--- a/include/configs/armada8k.h
+++ b/include/configs/armada8k.h
@@ -21,8 +21,8 @@
 #define __ARMADA_8K_H
 
 /* Version */
-#define CONFIG_IDENT_STRING             " armada8021-pxp"
-#define CONFIG_BOOTP_VCI_STRING         "U-boot.armv8.armada8021-pxp"
+#define CONFIG_IDENT_STRING             " Armada8k"
+#define CONFIG_BOOTP_VCI_STRING         "U-boot.armv8.armada8k"
 
 /* Enable IO drivers */
 #define MV_INCLUDE_UART
@@ -46,28 +46,28 @@
 
 #define CONFIG_AURORA_2_TIMER
 #define CONFIG_REMAKE_ELF
-#define CONFIG_SYS_ICACHE_OFF
-/* Generic Timer Definitions */
+
+/* Clock Definitions */
 #define COUNTER_FREQUENCY               (1008000)     /* 24MHz */
 
+/*
+ * Include the generic MVEBU configuration file.
+ * The generic file is affected by the defines above.
+ */
 #include "mvebu-common.h"
 
+/*
+ * SOC specific overrides
+ * Add here all config values that differ
+ * from the generic value in mvebu-common.h
+ */
 #undef CONFIG_BOOTDELAY
 #define CONFIG_BOOTDELAY                -1
 
 #undef CONFIG_BAUDRATE
 #define CONFIG_BAUDRATE                 63000
 
-/* Initial environment variables */
-#define CONFIG_EXTRA_ENV_SETTINGS	\
-					"kernel_addr=0x200000\0"	\
-					"initrd_addr=0xa00000\0"	\
-					"initrd_size=0x2000000\0"	\
-					"fdt_addr=0x100000\0"		\
-					"fdt_high=0xa0000000\0"
-
-#define CONFIG_BOOTARGS			"console=ttyAMA0 root=/dev/ram0"
-#define CONFIG_BOOTCOMMAND		"bootm $kernel_addr " \
-					"$initrd_addr:$initrd_size $fdt_addr"
+#undef CONFIG_BOOTARGS
+#define CONFIG_BOOTARGS			"console=ttyS0,63000 earlyprintk root=/dev/ram0"
 
 #endif /* __ARMADA_8K_H */
diff --git a/include/configs/mvebu-common.h b/include/configs/mvebu-common.h
index 1e0172b..0fde7c7 100644
--- a/include/configs/mvebu-common.h
+++ b/include/configs/mvebu-common.h
@@ -70,6 +70,7 @@
 
 /* Cache */
 #define CONFIG_SYS_DCACHE_OFF
+#define CONFIG_SYS_ICACHE_OFF
 #define CONFIG_SYS_CACHELINE_SIZE       32
 
 /*
@@ -87,7 +88,6 @@
 #define CONFIG_ZERO_BOOTDELAY_CHECK
 #define CONFIG_SYS_LOAD_ADDR		0x02000000      /* default load address	*/
 #define CONFIG_SYS_MIN_HDR_DEL_SIZE	0x100
-#undef  CONFIG_BOOTARGS
 
 /* Recovery */
 #ifdef MV_INCLUDE_RCVR
@@ -112,6 +112,16 @@
 #define CONFIG_SYS_BARGSIZE             CONFIG_SYS_CBSIZE       /* Boot Argument Buffer Size */
 #define CONFIG_BOOTDELAY                3
 #define CONFIG_ROOTPATH                 "/srv/nfs/"             /* Default Dir for NFS */
+#define CONFIG_EXTRA_ENV_SETTINGS	"kernel_addr=0x200000\0"	\
+					"initrd_addr=0xa00000\0"	\
+					"initrd_size=0x2000000\0"	\
+					"fdt_addr=0x100000\0"		\
+					"fdt_high=0xa0000000\0"
+
+#define CONFIG_BOOTARGS			"console=ttyS0,115200 earlyprintk root=/dev/ram0"
+#define CONFIG_BOOTCOMMAND		"bootm $kernel_addr $initrd_addr:$initrd_size $fdt_addr"
+
+/* Marvell specific env*/
 #define MV_BOOTARGS_END			":10.4.50.254:255.255.255.0:Marvell:eth0:none"
 #define MV_BOOTARGS_END_SWITCH		":::Marvell:eth0:none"
 
-- 
1.9.1

